File: Codeforces/A_141.cpp
// solved: 2013-11-29 14:11:53
#include <iostream>
#include <string>

using namespace std;
int main()
{
    string s;
    int let[26] = {0};
    cin >> s;
    for (int i = 0; i < s.length(); i++)
    {
        let[s[i] - 65]++;
    }
    cin >> s;
    for (int i = 0; i < s.length(); i++)
    {
        let[s[i] - 65]++;
    }
    int pile[26] = {0};
    cin >> s;
    for (int i = 0; i < s.length(); i++)
    {
        pile[s[i] - 65]++;
    }
    int i = 0;
    while (pile[i] == let[i])
        i++;
    if (i == 26)
        cout << "YES" << endl;
    else
        cout << "NO" << endl;
    return 0;
}

File: Codeforces/A_141.java
import java.util.*;


import java.lang.*;
import java.math.BigInteger;
import java.io.*;
public class A_141{
	public static void main (String[] args) throws java.lang.Exception{
		Scanner scan = new Scanner(System.in);
		String guest = scan.nextLine();
		String host = scan.nextLine();
		String pile = scan.nextLine();
		int[] pileLet = new int[26];
		int[] namesLet = new int[26];
		Arrays.fill(pileLet, 0);
		Arrays.fill(namesLet, 0);
		host += guest;
		for(char c : host.toCharArray()){
			namesLet[Character.getNumericValue(c) - Character.getNumericValue('A')]++;
		}
		for(char c : pile.toCharArray()){
			pileLet[Character.getNumericValue(c) - Character.getNumericValue('A')]++;
		}
		if(Arrays.toString(namesLet).equals(Arrays.toString(pileLet)))
			System.out.println("YES");
		else
			System.out.println("NO");
		scan.close();
	}
}